### ISSUE TITLE
[a11y]: Screen reader not announcing Error messages in DatePicker component

### ISSUE THREAD

**Issue by saiteja-nizam** [carbon-design-system/carbon#21986]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.102.0

### Automated testing tool and ruleset

MacOS voiceOver

### Description

Error messages associated with time-related input fields (such as start time, end time, or duration) are not being announced by screen readers when validation fails.

### Reproduction/example

@carbon/react

### Steps to reproduce

Open MacOS voice over and click on the `DatePicker` component and select a date such that an error occurs. Voice over is not announcing the error message.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Hi, could you reproduce the issue in a [stackblitz](https://stackblitz.com/github/carbon-design-system/sandboxes/tree/main/react/latest/react-19-vite?file=src%2Findex.scss,src%2FApp.jsx&preset=node=), and post the link back here? It's hard to know exactly what's going wrong without seeing the issue firsthand.

**Comment by saiteja-nizam** [comment]
> Hi, could you reproduce the issue in a [stackblitz](https://stackblitz.com/github/carbon-design-system/sandboxes/tree/main/react/latest/react-19-vite?file=src%2Findex.scss,src%2FApp.jsx&preset=node=), and post the link back here? It's hard to know exactly what's going wrong without seeing the issue firsthand.

https://github.com/user-attachments/assets/68c6f9c9-29c0-49be-bf43-8e22b3ae3a68

Screen reader not announcing the error message.
